File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    leaves = [0]
    
    for num in numbers:
        tmp =[]
        #leaves는 이전의 노드
        for leaf in leaves:
            tmp.append(leaf-num)
            tmp.append(leaf+num)
            
        leaves = tmp
            
    answer = leaves.count(target)        
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    int answer = 0;
    public int solution(int[] numbers, int target) {
        dfs(numbers,0,target,0);
        return answer;
    }
    
    public void dfs(int[]numbers,int depth, int target, int result){
        if (depth == numbers.length){
            if(result == target){
                answer ++;
            }
            return;
        }
        int plus = result + numbers[depth];
        int minus = result - numbers[depth];
        
        dfs(numbers,depth+1,target,plus);
        dfs(numbers,depth+1,target,minus);
    }
}